src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1100,8 +1100,12 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         // selection.
         if (_terminal->IsSelectionActive() && !KeyEvent::IsModifierKey(vkey) && vkey != VK_SNAPSHOT)
         {
-            _terminal->ClearSelection();
-            _renderer->TriggerSelection();
+            // GH#8791 - don't dismiss selection if Windows key was also pressed as a key-combination.
+            if (!(GetKeyState(VK_LWIN) & 0x8000) && !(GetKeyState(VK_RWIN) & 0x8000))

```

Comment:
Should this use [CoreWindow.GetKeyState](https://docs.microsoft.com/uwp/api/windows.ui.core.corewindow.getkeystate?view=winrt-19041) rather than [GetKeyState](https://docs.microsoft.com/windows/win32/api/winuser/nf-winuser-getkeystate), like TermControl::_GetPressedModifierKeys does? <https://github.com/microsoft/terminal/blob/d3c0d55aa5f13ca4e04d17e53c373ed6ec755a21/src/cascadia/TerminalControl/TermControl.cpp#L2915>

Reply:
Thanks! I think you're right. `CoreWindow.GetKeyState` also allows for better abstraction & better code-clarity, so I'll make those changes.